File: wash_away/lib/models/draft_booking_model.dart
class DraftBooking {
  final String? id;
  final int step;
  final String? serviceId;
  final String? vehicleTypeId;
  final String? vehicleTypeName;
  final DateTime? selectedDate;
  final String? selectedTime;
  final String? address;
  final String? additionalLocation;
  final String? paymentMethod;
  final String? couponCode;
  final DateTime? lastUpdated;

  DraftBooking({
    this.id,
    required this.step,
    this.serviceId,
    this.vehicleTypeId,
    this.vehicleTypeName,
    this.selectedDate,
    this.selectedTime,
    this.address,
    this.additionalLocation,
    this.paymentMethod,
    this.couponCode,
    this.lastUpdated,
  });

  factory DraftBooking.fromJson(Map<String, dynamic> json) {
    return DraftBooking(
      id: json['_id'] ?? json['id'],
      step: json['step'] ?? 0,
      serviceId: json['service_id']?['_id'] ?? json['service_id'],
      vehicleTypeId: json['vehicle_type_id']?['_id'] ?? json['vehicle_type_id'],
      vehicleTypeName: json['vehicle_type_name'] ?? json['vehicle_type_id']?['display_name'],
      selectedDate: json['selected_date'] != null 
          ? DateTime.parse(json['selected_date'])
          : null,
      selectedTime: json['selected_time'],
      address: json['address'],
      additionalLocation: json['additional_location'],
      paymentMethod: json['payment_method'],
      couponCode: json['coupon_code'],
      lastUpdated: json['updated_date'] != null
          ? DateTime.parse(json['updated_date'])
          : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'step': step,
      if (serviceId != null) 'service_id': serviceId,
      if (vehicleTypeId != null) 'vehicle_type_id': vehicleTypeId,
      if (vehicleTypeName != null) 'vehicle_type_name': vehicleTypeName,
      if (selectedDate != null) 'selected_date': selectedDate!.toIso8601String().split('T')[0],
      if (selectedTime != null) 'selected_time': selectedTime,
      if (address != null) 'address': address,
      if (additionalLocation != null) 'additional_location': additionalLocation,
      if (paymentMethod != null) 'payment_method': paymentMethod,
      if (couponCode != null) 'coupon_code': couponCode,
    };
  }
}

File: backend/src/models/DraftBooking.model.js
/* DRAFT BOOKING FUNCTIONALITY COMMENTED OUT
import mongoose from 'mongoose';

const draftBookingSchema = new mongoose.Schema({
  customer_id: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true,
    index: true
  },
  step: {
    type: Number,
    required: true,
    min: 1,
    max: 4
  },
  service_id: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Service'
  },
  vehicle_type_id: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'VehicleType'
  },
  vehicle_type_name: {
    type: String,
    trim: true
  },
  selected_date: {
    type: Date
  },
  selected_time: {
    type: String,
    trim: true
  },
  address: {
    type: String,
    trim: true
  },
  additional_location: {
    type: String,
    trim: true
  },
  payment_method: {
    type: String,
    enum: ['cash', 'card', 'wallet', 'apple_pay', 'google_pay'],
    trim: true
  },
  coupon_code: {
    type: String,
    trim: true
  },
  expires_at: {
    type: Date,
    default: () => new Date(Date.now() + 24 * 60 * 60 * 1000) // 24 hours from now
  },
  created_date: {
    type: Date,
    default: Date.now
  },
  updated_date: {
    type: Date,
    default: Date.now
  }
}, {
  timestamps: { createdAt: 'created_date', updatedAt: 'updated_date' }
});

// Index for faster queries
draftBookingSchema.index({ customer_id: 1 });
draftBookingSchema.index({ expires_at: 1 });

// Auto-delete expired drafts
draftBookingSchema.index({ expires_at: 1 }, { expireAfterSeconds: 0 });

const DraftBooking = mongoose.model('DraftBooking', draftBookingSchema);

export default DraftBooking;
*/
